File: Partition-Array-According-to-Given-Pivot.py
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        ans1 = []
        ans2 = []
        ans3 = []

        for num in nums:
            if num < pivot:
                ans1.append(num)
            
            elif  num > pivot:
                ans3.append(num)

            else:
                ans2.append(num)

        return ans1 + ans2 + ans3

File: Partition-Array-According-to-Given-Pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        int n = nums.size();
        vector<int> ans1;
        vector<int> ans2;
        vector<int> ans3;

        for(int i = 0; i < n; i++){
            if(nums[i] < pivot)
                ans1.push_back(nums[i]);
            
            else if(nums[i] > pivot)
                ans3.push_back(nums[i]);
            
            else
                ans2.push_back(nums[i]);
        }

        vector<int> ans;
        for(int i : ans1)
            ans.push_back(i);

        for(int i : ans2)
            ans.push_back(i);

        for(int i : ans3)
            ans.push_back(i);

        return ans;

    }
};